pkg/acsengine/defaults.go

```diff
@@ -80,6 +80,17 @@ func setOrchestratorDefaults(cs *api.ContainerService) {
 		if a.OrchestratorProfile.KubernetesConfig.NetworkPolicy == "" {
 			a.OrchestratorProfile.KubernetesConfig.NetworkPolicy = DefaultNetworkPolicy
 		}
+
+		if a.OrchestratorProfile.KubernetesConfig.DnsServiceIP == "" {
+			a.OrchestratorProfile.KubernetesConfig.DnsServiceIP = DefaultKubernetsDnsServiceIP
+		}
+		if a.OrchestratorProfile.KubernetesConfig.ServiceCIDR == "" {
+			a.OrchestratorProfile.KubernetesConfig.ServiceCIDR = DefaultKubernetsServiceCIDR
```

Comment:
To make the validation a code nicer I'm going to mandate that dnsServiceIp and serviceCIDR have to both be specified if either is. 

(Otherwise the validation code would have to know about the default values - validation is currently done on the versioned API model, which hasn't yet had the defaults applied.)